File: src/main/java/com/teamwizardry/librarianlib/gui/components/ComponentVoid.java
package com.teamwizardry.librarianlib.gui.components;

import com.teamwizardry.librarianlib.gui.GuiComponent;
import com.teamwizardry.librarianlib.math.Vec2d;

public class ComponentVoid extends GuiComponent<ComponentVoid> {

	public ComponentVoid(int posX, int posY, int width, int height) {
		super(posX, posY, width, height);
	}
	
	public ComponentVoid(int posX, int posY) {
		super(posX, posY);
	}

	@Override
	public void drawComponent(Vec2d mousePos, float partialTicks) {
		// NOOP
	}

}

File: src/main/java/com/teamwizardry/librarianlib/gui/components/ComponentVoid.kt
package com.teamwizardry.librarianlib.gui.components

import com.teamwizardry.librarianlib.gui.GuiComponent
import com.teamwizardry.librarianlib.math.Vec2d

class ComponentVoid : GuiComponent<ComponentVoid> {

    constructor(posX: Int, posY: Int, width: Int, height: Int) : super(posX, posY, width, height) {
    }

    constructor(posX: Int, posY: Int) : super(posX, posY) {
    }

    override fun drawComponent(mousePos: Vec2d, partialTicks: Float) {
        // NOOP
    }

}
